labonneboite/scripts/prepare_mailing_data.py

```diff
@@ -0,0 +1,163 @@
+    [
+        'office%s_%s' % (1+office_index, column_name)
+            for office_index in range(OFFICES_PER_USER)
+            for column_name in COLUMNS_FOR_EACH_OFFICE
+    ]
+
+# distances designed for Corsica
+DISTANCE_ATTEMPTS = [10, 20, 30]
+
+# tracking Google Analytics
+GA_TRACKING_SEARCH = 'utm_medium=email&utm_source=mailing_corse_201710&utm_campaign=lien_recherche'
+GA_TRACKING_DETAIL = 'utm_medium=email&utm_source=mailing_corse_201710&utm_campaign=lien_page_entreprise'
+
+
+def get_results(commune_id, rome_1_id, rome_2_id):
+    """
+    Try to find enough offices matching this commune_id,
+    for either of the two provided rome_ids,
+    and within as small of a distance as possible.
+
+    The result is a dictionary:
+    - offices found
+    - distance which was eventually used for the result
+    - rome_id which was eventually used for the result
+    """
+    city = geocoding.get_city_by_commune_id(commune_id)
+
+    if city is None:
+        return {'offices': [], 'distance': settings.DISTANCE_FILTER_DEFAULT, 'rome_id': rome_1_id}
+
+    latitude = city['coords']['lat']
+    longitude = city['coords']['lon']
+
+    mapper = mapping_util.Rome2NafMapper()
+
+    offices = []
+
+    # the only case where those stay None is if there is no result at all
+    used_distance = None
+    used_rome_id = None
+
+    for distance in DISTANCE_ATTEMPTS:
+        for rome_id in [rome_1_id, rome_2_id]:
+            if mapping_util.Rome2NafMapper.rome_is_valid(rome_id):
+                naf_code_list = mapper.map([rome_id])
+                offices, _, _ = fetch_companies(
+                    naf_codes=naf_code_list,
+                    rome_code=rome_id,
+                    latitude=latitude,
+                    longitude=longitude,
+                    distance=distance,
+                    from_number=1,
+                    to_number=OFFICES_PER_USER,
+                )
+                if len(offices) >= 1:
+                    used_rome_id = rome_id
+                    used_distance = distance
+                if len(offices) >= OFFICES_PER_USER:
+                    break
+        # FIXME any idea how to make this more DNRY ?
+        if len(offices) >= 1:
+            used_rome_id = rome_id
+            used_distance = distance
+        if len(offices) >= OFFICES_PER_USER:
+            break
+
+    return {'offices': offices, 'distance': used_distance, 'rome_id': used_rome_id}
+
+
+def get_search_url(commune_id, distance, rome_id):
+    """
+    Build search URL for given commune_id, distance and rome_id
+    """
+    if mapping_util.Rome2NafMapper.rome_is_valid(rome_id):
+        rome_description = settings.ROME_DESCRIPTIONS[rome_id.upper()]
+        slugified_rome_description = slugify(rome_description)
+        city = geocoding.get_city_by_commune_id(commune_id)
+        if city:
+            search_url = 'https://labonneboite.pole-emploi.fr/entreprises/%s-%s/%s?d=%s&%s' % (
```

Comment:
L'url est hard-coded ? Je crois que l'host est présent dans les settings, non ? Du coup il serait possible de générer cette url avec des appels à `HOST + url_for(...)` (ou un truc comme ça). De sorte que 

1. les résultats seront différents d'un environnement à un
2. `url_for` nous permet éventuellement de changer les urls un jour (oui, je sais, c'est un scénario fictif mais bon)

Reply:
en fait non car typiquement je fais tourner ce script en recette (pour pouvoir itérer facilement) et je veux obtenir les URLs de prod.   le jeu de données complet hyper confidentiel ne sort pas de la recette, on évite d'aller le mettre en prod. je vais rajouter un commentaire